_analyzers/character-filters/pattern-replace-character-filter.md

```diff
@@ -0,0 +1,238 @@
+
+The following request standardizes phone numbers by removing spaces, dashes, and parentheses: 
+
+```json
+GET /_analyze
+{
+  "tokenizer": "standard",
+  "char_filter": [
+    {
+      "type": "pattern_replace",
+      "pattern": "[\\s()-]+",
+      "replacement": ""
+    }
+  ],
+  "text": "(555) 123-4567"
+}
+```
+{% include copy-curl.html %}
+
+The response contains the generated token:
+
+```json
+{
+  "tokens": [
+    {
+      "token": "5551234567",
+      "start_offset": 1,
+      "end_offset": 14,
+      "type": "<NUM>",
+      "position": 0
+    }
+  ]
+}
+```
+ 
+## Parameters
+
+The html_strip character filter must be configured with the following parameters.
+
+| Parameter   | Required/Optional | Data type | Description    |
+|:---|:---|
+| `pattern`   | Required | String | A regular expression used to match parts of the input text. The filter identifies and matches this pattern for replacement. |
+| `replacement` | Optional | String | The string that replaces pattern matches. Use an empty string (`""`) to remove the matched text. Default is an empty string (`""`).   |
+
+## Creating a custom analyzer
+
+The following request creates an index with a custom analyzer configured with a `pattern_replace` character filter. The filter removes currency signs and thousands separators (both European `.` and American `,`) from numbers:
+
+```json
+PUT /my_index
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "my_analyzer": {
+          "tokenizer": "standard",
+          "char_filter": [
+            "pattern_char_filter"
+          ]
+        }
+      },
+      "char_filter": {
+        "pattern_char_filter": {
+          "type": "pattern_replace",
+          "pattern": "[$€,.]",
+          "replacement": ""
+        }
+      }
+    }
+  }
+}
+```
+
+{% include copy-curl.html %}
+
+Use the following request to examine the tokens generated using the analyzer:
+
+```json
+POST /my_index/_analyze
+{
+  "analyzer": "my_analyzer",
+  "text": "Total: $ 1,200.50 and € 1.100,75"
+}
+```
+{% include copy-curl.html %}
+
+The response contains the generated tokens:
+
+```json
+{
+  "tokens": [
+    {
+      "token": "Total",
+      "start_offset": 0,
+      "end_offset": 5,
+      "type": "<ALPHANUM>",
+      "position": 0
+    },
+    {
+      "token": "120050",
+      "start_offset": 9,
+      "end_offset": 17,
+      "type": "<NUM>",
+      "position": 1
+    },
+    {
+      "token": "and",
+      "start_offset": 18,
+      "end_offset": 21,
+      "type": "<ALPHANUM>",
+      "position": 2
+    },
+    {
+      "token": "110075",
+      "start_offset": 24,
+      "end_offset": 32,
+      "type": "<NUM>",
+      "position": 3
+    }
+  ]
+}
+```
+
+## Using capturing groups
+
+You can use capturing groups in `replacement`. For example, the following request creates a custom analyzer that uses a `pattern_replace` character filter to replace hyphens with dots in phone numbers:
```

Comment:
"in `replacement`" => "with the `replacement` parameter"?

Reply:
"in" is better. Reworded.